src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -30,9 +32,30 @@
 (def ServicesStatus
   {schema/Str ServiceStatus})
 
+(def SemVerVersion
+  (schema/pred semver/valid-format? "semver"))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
+(schema/defn ^:always-validate
+  get-service-version :- SemVerVersion
+  [group-id artifact-id]
+  (let [version (versioneer/get-version group-id artifact-id)]
+    (when-not version
+      (throw (IllegalStateException.
+               (format "Unable to find version number for '%s/%s'"
+                       group-id
+                       artifact-id))))
+    (when-not (semver/valid-format? version)
```

Comment:
Is the schema validation necessary given that we're doing the same validation manually here?